data/migrations/20190719081542_projects.js

```diff
@@ -0,0 +1,67 @@
+
+exports.up = function(knex) {
+    return knex.schema.createTable('projects', tbl => {
+        tbl.increments()
+  
+        tbl
+          .string('project_name')
+          .notNullable()
+          .unique()
+
+        tbl
+          .string('project_description')
+          .notNullable()   
+    })
+
+    .createTable('actions', tbl => {
+        tbl.increments()
+        
+        tbl
+            .string('action_description')
+            .notNullable()
+    
+        tbl
+            .string('action_notes')
+            .notNullable()
+
+        tbl
+            .integer('project_id')
+            .unsigned()
+            .notNullable()
+            .references('id')
+            .inTable('projects')
+            .onDelete('RESTRICT')
+            .onUpdate('CASCADE')
+    })
+
+    .createTable('project_actions', tbl => {
```

Comment:
Ok I see what you did here.... now technically this third relationship table, which tells me there's a many to many relationship, would be out of scope for this project. Though you can make a case for it if you remove the foreign key in the `actions` table. Here's a question you can ask yourself to help identify whether this table is required or not: is an action unique to one project or can I use one action for many projects?